src/main/java/org/apache/sling/adapter/internal/AdapterFactoryDescriptorMap.java

```diff
@@ -41,4 +42,16 @@
 
     private static final long serialVersionUID = 2L;
 
+    public AdapterFactoryDescriptorMap() {
+        super(new Comparator<ServiceReference<AdapterFactory>>() {
```

Comment:
Can you just use https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html#reversed--?

Reply:
this seems to be java8 only?

Reply:
Sorry, you are right. I guess in this module we still need to support older Java versions.